src/webots/nodes/WbRobot.cpp

```diff
@@ -1160,6 +1172,23 @@ void WbRobot::writeAnswer(QDataStream &stream) {
 
   if (mSupervisorUtilities)
     mSupervisorUtilities->writeAnswer(stream);
+
+  if (mNeedToWriteUrdf) {
+    stream << (short unsigned int)0;
+    stream << (unsigned char)C_ROBOT_URDF;
+
+    QString urdfContent;
+    WbVrmlWriter writer(&urdfContent, modelName() + ".urdf");
+
+    writer.writeHeader(name());
+    write(writer);
+    writer.writeFooter();
+
+    const QByteArray n = writer.readAll().toUtf8();
+    stream.writeRawData(n.constData(), n.size() + 1);
```

Comment:
I think I simplified now. Unfortunately, `stream << urdfContent` doesn't work, it is still uses `writeRawData()`

Reply:
Ok that seems strange because in the convert to base node we can successfully use directly the string, have you tried printing urdfContent? What does it contains?